### PR TITLE
Makisheva/UI tests

### PR DESCRIPTION
**Pull request by AlbinaMakisheva**

### CODE REVIEW

**Review by mirekys — DISMISSED** [review]

**Review by mirekys — CHANGES_REQUESTED** [review]

**Review by mirekys — CHANGES_REQUESTED** [review]
Fix sonar issues

File: tests/ui/util.js
Comment (mirekys) [inline tests/ui/util.js:1]: Why do all of this crypto-dance when only a simple random number is needed? 😅

**Review by mirekys — CHANGES_REQUESTED** [review]

File: tests/ui/util.js
Comment (AlbinaMakisheva) [reply tests/ui/util.js:1]: Math.random caused issues in sonar with high level of danger

File: tests/ui/create-form.spec.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      .locator("visible=true")
+      .count();
+
+    const randomIndex = getRandomInt(numberOfOptions);
+
+    await page
+      .locator(".tree-column .row")
+      .locator("visible=true")
+      .nth(randomIndex)
+      .dblclick();
```
Comment (mirekys) [inline tests/ui/create-form.spec.ts:41]: Now will be single-click

File: tests/ui/dashboard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+test("get records", async ({ page, baseURL }) => {
+  await page.goto("/me/records");
+  await page.locator("#invenio-burger-toggle").click();
+  await page.getByTestId("dashboard-redir-btn").click();
+
+  await expect(page).toHaveURL(
+    `${baseURL}me/records/?q=&l=list&p=1&s=10&sort=newest`
+  );
+
+  await expect(page.getByTestId("aggregation-count")).toContainText("20");
```
Comment (mirekys) [inline tests/ui/dashboard.spec.ts:120]: Where does that 20 records come from?

**Review by mirekys — CHANGES_REQUESTED** [review]

File: tests/ui/util.js
Comment (AlbinaMakisheva) [reply tests/ui/util.js:1]: https://rules.sonarsource.com/javascript/tag/cwe/RSPEC-2245/

Comment (mirekys) [reply tests/ui/util.js:1]: Is it really a security risk, when someone could (given some non-trivial effort) guess which item will be randomly clicked on, while the test scenario is executed by CI?

File: tests/ui/dashboard.spec.ts (hunk 2 above)
Comment (mirekys) [reply tests/ui/dashboard.spec.ts:119]: What is this "aggregation-count" element? Is it count of items in some aggregation value? Or result count?

**Review by mirekys — DISMISSED** [review]

**Comment by AlbinaMakisheva** [comment]
@mirekys I fixed a conflict with main, so one more approval is required

**Pull request merged**